### PR TITLE
Init Tests

### PR DESCRIPTION
**Pull request by roramirez**

### CODE REVIEW

**Comment by MichMich** [comment]
Does this mean you're going to work on a full test suite?

**Comment by roramirez** [comment]
Hi @MichMich 

I've interest a work in a test suite for MagicMirror. I think is somethings we need.

I can do develop somes tests. This patch propose the use mocha for the test suite.

I would like to hear comments and if  we can work a list set of test to do.

**Comment by MichMich** [comment]
Honestly, I don't have much experience with writing test. This would be a nice project to learn more.

**Comment by roramirez** [comment]
We can invite to users in forums. I think is really important define the test list for will do. The test suite is a little big work, isn't something we can do it in one day or week.

**Comment by qistoph** [comment]
One of the changes in 3a8d72d introduces an error in the client.

[js/module.js:419](https://github.com/MichMich/MagicMirror/commit/3a8d72db313e62729dbb6f9061415ff615b7ba0d#diff-ea3e1f30bd2d56ae767ccdfcd88fba8eR419)

`Uncaught ReferenceError: exports is not defined
    at module.js:419`

**Comment by roramirez** [comment]
I see. Let me check how prevent this. If you have any suggest or aprouch is welcome :)

File: js/module.js [hunk 1]
```diff
@@ -415,3 +415,7 @@ Module.register = function (name, moduleDefinition) {
 	Log.log("Module registered: " + name);
 	Module.definitions[name] = moduleDefinition;
 };
+
+exports._test = {
```
Comment (nhubbard) [inline js/module.js]: I think that you would do this instead:
```javascript
module.exports = {
    cmpVersions: cmpVersions
}
```

Comment (roramirez) [inline js/module.js]: This isn't work :(. I sent a new Pull Request with a hotfix. Can you review @nhubbard and @qistoph this?

Also, I add a post for create a set list of test for the suite. 
https://forum.magicmirror.builders/topic/1456/test-suite-for-magicmirror
I would really like have your comments.

Comment (qistoph) [inline js/module.js]: PR #633 removed the error for me and `npm test` works. Seems like a good solution to me.

Also noticed a minor textual improvement in `tests/functions/compare-version.js`:
"Test function ... **into** ..." could better be "Test function ... **in** ...".

Comment (roramirez) [inline js/module.js]: Thanks. I pushed the textual change.

**Pull request merged**